src/deferred_data.rs

```diff
@@ -102,6 +112,17 @@ impl<T: DeferredDataSource> CountingDeferredDataSource<T> {
 }
 
 impl<T: DeferredDataSource> DeferredDataSource for CountingDeferredDataSource<T> {
+    fn fetch_description(&mut self) {
+        self.start_request();
+        self.data_source.fetch_description()
+    }
+
+    fn get_description(&mut self) -> DataSourceDescription {
+        let result = self.data_source.get_description();
+        self.outstanding_requests -= 1;
```

Comment:
could add a similar assert here as `finish_request` or add a new `finish_request` func for non-vec types

Reply:
See my last comment. If this is a network request, we must track it as such. If not, then I'd get rid of this entirely and not touch `outstanding_requests` at all for the `get_description` method.

Reply:
https://github.com/StanfordLegion/prof-viewer/pull/36/commits/c28219e2ce499d54bda9492294abdbf073034ced